### ISSUE TITLE
Implement InstantNGP

### ISSUE THREAD

**Issue by MeepOwned13** [MeepOwned13/raymeshnerf#5]
- [InstantNGP paper](https://dl.acm.org/doi/10.1145/3528223.3530127)
- [InstantNGP repo](https://github.com/NVlabs/instant-ngp)
- [InstantNGP PyTorch implementation](https://github.com/cheind/pure-torch-ngp/blob/develop/torchngp/modules/nerf.py)

**Comment by MeepOwned13** [comment]
Progress up until now:
- InstanNGP implemented
- ImportantPixelSampling had to be removed because of too many rays (weighted sampling can't take more than 2^24 values)

TODO list before next training attempt:
- Backgrounds should be treated differently on loss calculation, add small noise to color to force 0 density (renderer needs to return depths for this)
- Sample rays in batch from a single image, handling it this way allows ImportantPixelSampling to be used per image (maybe images can be sampled by sum of sampler weights)
- LRScheduler could be more aggressive, maybe define some epsilon
